### DISCUSSION TITLE
how to build and test imodel-native repository

### DISCUSSION THREAD

**Discussion by QianSunCN** [iTwin/itwinjs-core#5813] [Q&A]
iTwin/imodel-native

hi, please could you help me how to build and test this repository, I didn't find a readme or guide for this repository. 
Thanks!

**Comment by pmconne** [comment]
Hi,

We [should](https://github.com/iTwin/imodel-native/issues/30) add a README to the repository root. The [iModelJsNodeAddon README](https://github.com/iTwin/imodel-native/blob/main/iModelJsNodeAddon/README-Private.md) contains some information, but it won't be very useful to you. We use "bentley build" to build the addon, and we haven't gotten around to open-sourcing that yet, due in part to lack of demand.

At the moment, if you want to be able to build the addon, you'll need to write some build scripts (e.g., using cmake) to do it.

**Reply by QianSunCN** [reply]
thanks.
